src/cargo/util/config.rs

```diff
@@ -424,11 +424,25 @@ impl ConfigValue {
 }
 
 fn homedir() -> Option<PathBuf> {
-    let cargo_home = env::var_os("CARGO_HOME").map(PathBuf::from);
+    let cargo_home = get_cargo_home_dir();
     let user_home = env::home_dir().map(|p| p.join(".cargo"));
     return cargo_home.or(user_home);
 }
 
+fn get_cargo_home_dir() -> Option<PathBuf> {
+    let cargo_home = env::var_os("CARGO_HOME").map(PathBuf::from);
+    match cargo_home {
+        Some(path) => if !path.is_absolute() {
+            let mut absolute_path = env::current_dir().unwrap();
```

Comment:
We tend to try to avoid `unwrap` as much as possible in Cargo, could the `cwd` be an argument to the `homedir` function above which is passed down to this? I think it may also be fine to leave this in the `homedir` function as:

```rust
let cargo_home = env::var_os("CARGO_HOME").map(|home| {
    cwd.join(home)
});
```

The `join` function will just return `home` if it's already absolute I believe.